gossipsub.go

```diff
@@ -899,18 +899,91 @@ func (gs *GossipSubRouter) sendRPC(p peer.ID, out *RPC) {
 		return
 	}
 
-	select {
-	case mch <- out:
-		gs.tracer.SendRPC(out, p)
-	default:
-		log.Infof("dropping message to peer %s: queue full", p)
+	// fragment into multiple RPCs if necessary
+	outRPCs, err := fragmentRPC(out, gs.p.maxMessageSize)
+	if err != nil {
+		log.Warnf("unable to fragment message to peer %s: %s", p, err)
 		gs.tracer.DropRPC(out, p)
 		// push control messages that need to be retried
 		ctl := out.GetControl()
 		if ctl != nil {
 			gs.pushControl(p, ctl)
 		}
+		return
+	}
+
+	for _, rpc := range outRPCs {
+		select {
+		case mch <- rpc:
+			gs.tracer.SendRPC(rpc, p)
+		default:
+			log.Infof("dropping message to peer %s: queue full", p)
+			gs.tracer.DropRPC(rpc, p)
+			// push control messages that need to be retried
+			ctl := rpc.GetControl()
+			if ctl != nil {
+				gs.pushControl(p, ctl)
+			}
+		}
+	}
+}
+
+func fragmentRPC(rpc *RPC, limit int) ([]*RPC, error) {
```

Comment:
I think we should separate the control messages in their own RPC if we are over size -- are we counting those? we could overflow because of them.